lua/telescope/_extensions/papis/actions.lua

```diff
@@ -11,50 +11,108 @@ local db = require("papis.sqlite-wrapper")
 
 local utils = require("papis.utils")
 
+local get_multi = function(prompt_bufnr)
+  local picker = require('telescope.actions.state').get_current_picker(prompt_bufnr)
+  local multi = picker:get_multi_selection()
+  return multi
+end
+
 local M = {}
 
----This function inserts a formatted reference string at the cursor
+---This function inserts a formatted ref string at the cursor
 ---@param prompt_bufnr number @The buffer number of the prompt
 ---@param format_string string @The string to be inserted
 M.ref_insert = function(prompt_bufnr, format_string)
-  local entry = string.format(format_string, action_state.get_selected_entry().id.ref)
+  local multi = get_multi(prompt_bufnr)
+
   actions.close(prompt_bufnr)
-  vim.api.nvim_put({ entry }, "", false, true)
+  local string_to_insert = ""
+  if vim.tbl_isempty(multi) then
+    local ref = string.format(format_string, action_state.get_selected_entry().id.ref)
+    string_to_insert = ref
+  else
+    for _, entry in pairs(multi) do
+      local ref = string.format(format_string, entry.id.ref)
+      string_to_insert = string_to_insert .. ref .. " "
```

Comment:
The string connecting the reference keys (`" "` here)  could be another configuration parameter. I only use `papis.nvim` with `tex` files and I need to connect them via `,`

Reply:
Good point! I'll make sure to make it configurable and try to see what would be sensible defaults for the various filetypes we configure in `cite_formats`. I might in fact add it to that option :)